types/src/blockchain.rs

```diff
@@ -83,10 +83,16 @@ pub enum BCReadRequest {
     /// The input is a list of output amounts.
     NumberOutputsWithAmount(Vec<u64>),
 
-    /// Check that all key images within a set arer not spent.
+    /// Check that all key images within a set are not spent.
     ///
     /// Input is a set of key images.
     KeyImagesSpent(HashSet<[u8; 32]>),
+
+    /// A request for the compact chain history.
+    CompactChainHistory,
+
+    /// A request to find the first unknown block ID in a list of block IDs.
+    FindFirstUnknown(Vec<[u8; 32]>),
```

Comment:
```suggestion
    /// A request to find the first unknown block ID in a list of block IDs.
    ////
    /// # Invariant
    /// The [`Vec`] containing the block IDs must be sorted in chronological block
    /// order, or else the returned response is unspecified and meaningless,
    /// as this request performs a binary search.
    FindFirstUnknown(Vec<[u8; 32]>),
```

Reply:
Also the binary search isn't guaranteed to return the _first_ unknown block hash, right? Maybe this should be renamed.

Reply:
It finds the partition point between the known and the unknown blocks, so it does return the first